tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/dialogs/UIWorkingSetWizardsAuto.java

```diff
@@ -134,11 +137,24 @@ private void removeAllWorkingSets() {
 	}
 
 	private void cleanupWorkspace() {
+		IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
 		try {
-			ResourcesPlugin.getWorkspace().getRoot().delete(true, null);
+			root.delete(true, null);
 		} catch (CoreException e) {
-			TestPlugin.getDefault().getLog().log(e.getStatus());
-			throw createAssertionError(e);
+			// give it some more time
+			try {
+				TimeUnit.SECONDS.sleep(1);
+			} catch (InterruptedException e1) {
+			}
+			try {
+				root.refreshLocal(IResource.DEPTH_INFINITE, null);
+				if (root.exists()) {
+					root.delete(true, null);
+				}
+			} catch (CoreException e1) {
+				TestPlugin.getDefault().getLog().log(e.getStatus());
+				throw createAssertionError(e);
+			}
```

Comment:
Are these changes related to this PR? Judging by the description of the PR, it doesn't look like they do.

Reply:
The test seem to randomly fail so I added another commit here to see if it can be fixed.

Reply:
But the failing test doesn't have anything to do with `ApplicationPartServiceImpl` right?

Reply:
Only that it failed while building the PR once...

Reply:
I would propose to keep this separate and have another PR that is also linked to the according issue: #275

Reply:
Ah, thank you @HeikoKlare , I was looking for the right issue to link here :-)

@laeubi please propose another PR for these changes.